examples/deck/example.js

```diff
@@ -334,6 +341,60 @@ export default class Example extends Component<
     );
   }
 
+  _renderPolygonSnappingControls() {
+    const defaultSnapStrength = 0.2;
+    const snapStrengthValue =
+      (this.state.modeConfig && this.state.modeConfig.snapStrength) || defaultSnapStrength;
+    return (
+      <div key="translate">
+        <ToolboxRow>
+          <ToolboxLabel>Enable polygon snapping</ToolboxLabel>
+          <ToolboxControl>
+            <input
+              type="checkbox"
+              checked={Boolean(
+                this.state.modeConfig && this.state.modeConfig.enablePolygonSnapping
+              )}
+              onChange={event => {
+                const initModeConfig = this.state.modeConfig;
+                const { snapStrength } = initModeConfig;
+                const modeConfig = {
+                  ...initModeConfig,
+                  enablePolygonSnapping: Boolean(event.target.checked)
+                };
+                if (!snapStrength) {
+                  modeConfig.snapStrength = defaultSnapStrength;
+                }
+                this.setState({ modeConfig });
+              }}
+            />
+          </ToolboxControl>
+        </ToolboxRow>
+
+        <ToolboxRow>
+          <ToolboxLabel>Snap strength</ToolboxLabel>
+          <ToolboxControl>
+            <input
+              type="range"
+              min="0.1"
+              max="1"
+              step="0.1"
+              value={snapStrengthValue}
```

Comment:
I think rather than using an arbitrary "strength" unit. What if this is simply snap "pixels" and is used as the argument to picking?

Reply:
Sure this sounds like a good idea. Is there a way for me to project world latlong space to screen pixel space? I'm thinking of using a snap pixel threshold that remains constant throughout all zoom levels.

Reply:
You don't need to project to world space. Just use the `snapPixels` value as the `radius` argument to `pickObject`. Or you might need to call `pickMultipleObjects`, don't recall the exact method name for picking multiple. See `modules/layers/src/layers/editable-layer.js`. If `pickObject` returns a snappable edit handle (the ones that are gray in the demo), then you use turf and ground coordinates to get the closest point on that polygon to the point being dragged.

Reply:
How do I get access to the deck class's `pickMultipleObjects` method from within `mode-handler`? I can only find `pickObject` and `pickObjects` through `this._context.layerManager`.

Reply:
I think picks are already handed to the mode handlers. But you might need to tweak the interface to support multi-depth picking since you'll need the gray edit handle underneath the purple one. We might be able to hard-code the picking depth to `2` for now. But in the future, we might need some way for a mode handler to ask the layer to do deeper picks if necessary for that particular mode.